### ISSUE TITLE
Crash returning to page with speedreader enabled

### ISSUE THREAD

**Issue by rillian** [brave/brave-browser#24588]
IMPORTANT: Your crash has already been automatically reported to our crash system. Please file this bug only if you can provide more information about it.

Brave Version: 1.43.58  Chromium: 104.0.5112.81
Operating System: Linux 5.18.16-200.fc36.x86_64

URL (if applicable) where crash occurred: https://blog.rust-lang.org/2022/08/11/Rust-1.63.0.html

Can you reproduce this crash? No. Was reliable for a while, but currently not re-occuring.

What steps will reproduce this crash? (If it's not reproducible, what were you doing just before the crash?)
1. Visit https://blog.rust-lang.org/2022/08/11/Rust-1.63.0.html with speedreader enabled
2. Follow link to doc.rust-lang.org
3. Click back button to navigate back to the blog post
4. Tab reports segv

NB: Still occurs if I click 'Show original page' first.

****DO NOT CHANGE BELOW THIS LINE****
Crash ID: crash/1bfd0000-f3fb-a908-0000-000000000000

**Comment by rebron** [comment]
cc: @boocmp

**Comment by rillian** [comment]
:tada:

**Comment by kjozwiak** [comment]
Going to label this as `QA/No` as I couldn't reproduce the issue using the STR/Cases outlined above. Tried multiple times using `1.44.48 Chromium: 104.0.5112.81` which doesn't have https://github.com/brave/brave-core/pull/14642 and couldn't reproduce on either `Windows` & `Linux`. Double checked with @rillian and he mentioned it was intermittent.

However, I did quick sanity check and made sure `Speedreader` was working as expected. Some more context can be found via https://bravesoftware.slack.com/archives/CLX7EV42Z/p1660857603829149.
